src/flag_gems/utils/shape_utils.py

```diff
@@ -137,3 +139,30 @@ def dim_compress(inp, dims):
     sorted_reduction_dim = sorted(dims, key=lambda x: stride[x], reverse=True)
     order = batch_dim + sorted_reduction_dim
     return inp.permute(order).contiguous()
+
+
+def offsetCalculator(inp, idx, strides, dim, isInp):
+    ndim = inp.ndim
+    shape = list(inp.shape)
+    offsets = 0
+    idx_dim = 0
+    for d in range(0, ndim):
+        mod = floor(idx % shape[d])
+        add_on = mod * strides[d]
+        offsets += add_on
+        if d == dim:
+            idx_dim = add_on
+        idx = idx // shape[d]
+        # FIXME: Should we write a fast div/mod
```

Comment:
I think we do need a fast div-mod functor, but i wonder if it already exist in triton ？

Reply:
I checked out triton's source code and found that its fast division is implemented using NumPy's `np.divide`, and as far as I know, np's integer division is optimized. So do you think we just need to import np's div/mod operators in this file to use them?